### PR TITLE
fix #22168, `im*I+I` should be shown as `(1+1im)*I`

### LINKED ISSUE #22168: UniformScaling should use parentheses if the scalar is complex
entering `im*I+I` is shown as `1+1im*I` though it calculates as `(1+1im)*I` thus i guess that's some sort of wrong display algorithm.

(Since i've no clue of committing rules i thought that'd be a really easy to fix problem for you and can happen pretty much instant within your normal workflow ^^')

### PR DESCRIPTION
**Pull request by sunoru**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
Looks good to me. Thanks for the contribution!

File: base/linalg/uniformscaling.jl [hunk 1]
```diff
@@ -31,6 +31,7 @@ eltype(::Type{UniformScaling{T}}) where {T} = T
 ndims(J::UniformScaling) = 2
 getindex(J::UniformScaling, i::Integer,j::Integer) = ifelse(i==j,J.λ,zero(J.λ))
 
+show(io::IO, J::UniformScaling{T}) where {T<:Complex} = print(io, "$(typeof(J))\n($(J.λ))*I")
```
Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: Could we do a check for the number of terms instead? Minor use case but this would still be wrong for quaternions.

Comment (sunoru) [inline base/linalg/uniformscaling.jl]: That's true. But if we do such a check we have to deal with strings. Don't know if it's OK to do so.

Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: Aren't we already dealing with strings? My idea was to add the parentheses if something like `ismatch(r"[0-9]+ ?\+|\- ?[0-9]+", "$(J.λ)")`

Comment (sunoru) [inline base/linalg/uniformscaling.jl]: Oh I see. How about just checking `r"\w+\s*\+|\-\s*\w+"` as there must be `+` or `-` between terms if there are more than one term.

Comment (andreasnoack) [inline base/linalg/uniformscaling.jl]: That was also my first version but I think negative reals become false positives here
```julia
julia> ismatch(r"\w+\s*\+|\-\s*\w+", "-1")
true
```

Comment (sunoru) [inline base/linalg/uniformscaling.jl]: Ah I'm sorry. It's because of the misuse of `|` and we should use `r"\w+\s*[\+\-]\s*\w+"`.

Comment (KristofferC) [inline base/linalg/uniformscaling.jl]: Could you perhaps add tests for the things that wasn't completely right with the first regexp.

Comment (sunoru) [inline base/linalg/uniformscaling.jl]: I don't know what common things can be wrong with the first regexp, but I believe there could be situations when users define a type that doesn't contain digit-[plus/minus]-digit pattern, such as something like `1i+2j`. (The current displays of complex and quaternion numbers have something like `0+0im` so the first regexp always works with them.)

**Pull request merged**